testing/test_kerns.py

```diff
@@ -221,29 +221,38 @@ class TestSlice(unittest.TestCase):
     """
 
     def setUp(self):
-        tf.reset_default_graph()
         self.rng = np.random.RandomState(0)
-        self.k1 = GPflow.kernels.RBF(1, active_dims=[0])
-        self.k2 = GPflow.kernels.RBF(1, active_dims=[1])
-        self.k3 = GPflow.kernels.RBF(1)
+        tf.reset_default_graph()
+
         self.X = self.rng.randn(20, 2)
         self.Z = self.rng.randn(10, 2)
 
+        kernels = GPflow.kernels.Stationary.__subclasses__() + [GPflow.kernels.Constant, GPflow.kernels.Linear,
+                                                                GPflow.kernels.Polynomial]
+        self.kernels = []
+        for kernclass in kernels:
+            k1 = kernclass(self.X.shape[1], active_dims=[0])
```

Comment:
Or to put it another way, after `_slice`, `tf.shape(X)[1]` should be equal to `active_dims`? Should we place an assertion for this in the code? I feel like currently it's really brittle, that you can pass in a set of arguments that will result in the kernel working on a different number of dimensions. This is an issue especially for the kernel expectations, where certain constants depend on the number of dimensions the kernel is acting on.